cpp/src/arrow/filesystem/gcsfs.cc

```diff
@@ -267,6 +267,18 @@ google::cloud::Options AsGoogleCloudOptions(const GcsOptions& o) {
   return options;
 }
 
+class GcsCredentialsProvider {
```

Comment:
Nit, but it would probably be enough to make this a simple `struct` with a public member. The attribute wrapping doesn't seem very useful here.

Reply:
Done.